### ISSUE TITLE
Cannot Post Message

### ISSUE THREAD

**Issue by enniob** [mcwebb/angular-twilio#2]
I'm using your code, but when I use the example to send a message I get the following error.

Code: 21603
message: "A 'From' phone number is required."
more_info: "https://www.twilio.com/docs/errors/21603"
status: 400

Any idea why?

**Comment by mcwebb** [comment]
Are you definitely sending a `From` value?
When you send a message with Twilio you need to tell Twilio which of your numbers you want to use to send it.

``` javascript
Twilio.create('Messages', {
    From: '+12402926537',
    To: '+16175551212',
    Body: 'This is a test! yay!'
});
```

**Comment by ghost** [comment]
I have the same error :(

**Comment by mcwebb** [comment]
Hi @happschinson,
Can you confirm whether or not you're definitely sending a valid `From` value?

**Comment by ghost** [comment]
yep. tried it with a capital F as here in the comments as well as with a lower f

**Comment by mcwebb** [comment]
And you definitely _own_ that `from` number in Twilio?
Just want to make sure, thanks.

**Comment by ghost** [comment]
I'm on a trial account, but that's my number i received from twilio, yep

**Comment by mcwebb** [comment]
In your browser's web tools, could you open the Network panel and see what is recorded when you initiate the Twilio call.
Specifically what data is sent to Twilio's servers.

**Comment by RoryShively** [comment]
Yup I'm having the same issue
Definitely using the appropriate from number

**Comment by mcwebb** [comment]
Thanks for confirming @RoryShively, could you take the steps mentioned above  (https://github.com/mcwebb/angular-twilio/issues/2#issuecomment-149613952).
Need to know whether the `form` field is even being sent to Twilio's servers.

**Comment by RoryShively** [comment]
error: "unauthorized" 
That definitely doesn't help though :(

http://stackoverflow.com/questions/23617722/sending-sms-in-twilio-returns-21603-a-from-number-is-required
I have a feeling this post might help though. I'm going to try and work it out tonight and I'll let you know what I find

**Comment by mcwebb** [comment]
Very strange. Are you able to tell if there is any data being sent to their servers? esp. the `From` field.
Maybe try sending a message via cURL as shown in the last answer to that question. At least then we'd know whether or not this module in itself is the issue.

**Comment by RoryShively** [comment]
Yes the payload has From, To, and Body in it. 
From=%2B##########&To=%2B##########&Body=Message+Goes+Here

**Comment by mcwebb** [comment]
Okay thanks, then I'm sure why that's not working. I'll look into it.
And let me know if you dig anything up.
Thanks for your help.

**Comment by RoryShively** [comment]
http://stackoverflow.com/questions/25232751/cant-send-sms-message-with-test-credentials-and-rest-console-in-chrome
I haven't had a chance to check this out but url-encoded params might be the solution

**Comment by RoryShively** [comment]
I got it working, dont use json and your gravy

**Comment by firmanhidayat** [comment]
I have same problem, and then does it solve the problem?

**Comment by mcwebb** [comment]
It looks like @bcfchen has attempted to fix the issue, maybe try his fork.
https://github.com/bcfchen/angular-twilio
If it works for you, could you or bcfchen submit a PR? Thanks

**Comment by firmanhidayat** [comment]
It does not work for me 
these are message : {"code": 21603, "message": "A 'From' phone number is required.", "more_info": "https://www.twilio.com/docs/errors/21603", "status": 400}

**Comment by enniob** [comment]
I'm sorry for the long delay on posting here, I just have been busy with my projects. But I was never able to get this working and I always got the same error that everyone was getting. 

I was able to get it working with the following code.
`
    messageObj.sendMessage = function(phoneNumber,message){
      return $http({
        method:'post',
        url: twilio_url,
        headers: {'Content-Type': 'application/x-www-form-urlencoded',
          'Authorization' : 'Basic ' + $window.btoa('xxxxxxx:xxxxxxxxx')},
        transformRequest: function(obj) {
              var str = [];
              for(var p in obj)
              str.push(encodeURIComponent(p) + "=" + encodeURIComponent(obj[p]));
              return str.join("&");
        },
        data: {From:'YOUFROMNUMBER',To:phoneNumber,Body:message}
      }).success(function(resp){
         //DO SOMETHING
      });
    };
`

**Comment by HofmannZ** [comment]
I'm having the same issue, any idea how to fix it yet?

**Comment by mcwebb** [comment]
Hi @HofmannZ could you check out the PR #3.
if it works for you could you let me know and I'll merge it.

**Comment by HofmannZ** [comment]
Hey @mcwebb it does not seem to work. I temporarily use this JQuery plugin [JsFiddle by Dekkard](http://jsfiddle.net/dekkard/w5ordwwd/). Might be a bit of inspiration?

**Comment by mcwebb** [comment]
Thanks Hofmann, I'm not actively maintaining this module atm, but if I have time next month I'll have a look at the link and see if I can find the issue and patch it. In the meantime I'm very open to any PRs.

**Comment by mcwebb** [comment]
Closing since this should be fixed by #4.
Let me know if not.
